File: app/Sources/TarmacKit/BootTerminal.swift
import Foundation

/// Mints fresh, globally-unique `term_id`s for boot / ⌘T / restored-extra
/// terminals. Isolated in TarmacKit so terminal-id minting has a single home
/// (`AppController` calls `mint()` instead of scattering `UUID().uuidString`),
/// giving one place to change the scheme later (e.g. for deterministic ids).
public enum BootTerminal {
    /// A fresh globally-unique terminal id.
    public static func mint() -> String {
        UUID().uuidString
    }
}

File: desktop/src/kit/bootTerminal.ts
// Port of TarmacKit/BootTerminal.swift — the single home for terminal-id
// minting (boot / ⌘T / restored-extra terminals). Isolated so id minting lives
// in one place (callers use `mint()` instead of scattering UUID generation),
// giving one spot to change the scheme later (e.g. for deterministic ids).
// Swift used `UUID().uuidString`; the TS analogue is `crypto.randomUUID()`.

/** A fresh globally-unique terminal id. */
export function mint(): string {
  return crypto.randomUUID();
}
